From efd6409f825122b890a0a12fc35c42733e993b00 Mon Sep 17 00:00:00 2001
From: Dien Pham <dien.pham.ry@renesas.com>
Date: Thu, 12 Apr 2018 16:03:30 +0700
Subject: [PATCH 032/909] soc: renesas: rcar-sysc: Add SYSCEXTMASK info for
 r8a7796

commit ee953e64cc1338c98a60e686400425fc1f1cb8c0 from
git://git.kernel.org/pub/scm/linux/kernel/git/horms/renesas-bsp.git

Add specific SYSCEXTMASK register info so that
it is controlled during power domain on/off.

Note: as hardware team info, this update is not
necessary for M3 v1.*.

Signed-off-by: Dien Pham <dien.pham.ry@renesas.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/soc/renesas/r8a7796-sysc.c | 21 +++++++++++++++++++++
 1 file changed, 21 insertions(+)

diff --git a/drivers/soc/renesas/r8a7796-sysc.c b/drivers/soc/renesas/r8a7796-sysc.c
index f700c842b9e1..afab100287e8 100644
--- a/drivers/soc/renesas/r8a7796-sysc.c
+++ b/drivers/soc/renesas/r8a7796-sysc.c
@@ -8,8 +8,10 @@
  * the Free Software Foundation; version 2 of the License.
  */
 
+#include <linux/bitops.h>
 #include <linux/bug.h>
 #include <linux/kernel.h>
+#include <linux/sys_soc.h>
 
 #include <dt-bindings/power/r8a7796-sysc.h>
 
@@ -42,7 +44,26 @@ static const struct rcar_sysc_area r8a7796_areas[] __initconst = {
 	{ "a3ir",	0x180, 0, R8A7796_PD_A3IR,	R8A7796_PD_ALWAYS_ON },
 };
 
+static const struct soc_device_attribute r8a7796es1[] __initconst = {
+	{ .soc_id = "r8a7796", .revision = "ES1.*" },
+	{ /* sentinel */ }
+};
+
+static int __init r8a7796_sysc_init(void)
+{
+	if (soc_device_match(r8a7796es1))
+		r8a7796_sysc_info.extra_regs->sysc_extmask_msks = 0;
+
+	return 0;
+}
+
+static struct rcar_sysc_extra_regs r8a7796_extra_regs = {
+	.sysc_extmask_offs = 0x2F8, .sysc_extmask_msks = BIT(0)
+};
+
 const struct rcar_sysc_info r8a7796_sysc_info __initconst = {
+	.init = r8a7796_sysc_init,
 	.areas = r8a7796_areas,
 	.num_areas = ARRAY_SIZE(r8a7796_areas),
+	.extra_regs = &r8a7796_extra_regs,
 };
-- 
2.17.1

